scripts/pc2sandbox.sh

```diff
@@ -33,23 +33,40 @@ FAIL_SANDBOX_ERROR=$((FAIL_RETCODE_BASE+54))
 MAXPROCS=32
 
 # taskset cpu mask for running submission on single processor
-CPUMASK=0x01
+cpunum=${USER/judge/}
```

Comment:
It's probably due to the fact that my Bash knowledge has weakened over time, but I don't understand what the above line is doing.  Specifically, what is the value of the variable `USER/judge/`?    ( know what `${USER}` means; I don't know what the meaning of adding the `/judge/` to the argument is...)

And what happens if the submission is being executed by an Admin rather than a judge?

Reply:
It removes judge from the USER variable

Reply:
basically: cpunum=echo $USER | sed -e 's/judge//'
erases judge from the USER variable so it just gets the judge number.  This was added for running multiple judges on the same machine different judge unix login accounts, and pinning them to different CPU's.  If the account name is not "judge#", for example, it will use CPU 0, which is fine if only one judge is running (or admin) on the system.  We usually only run one client application at-a-time on a machine, so that's ok.

Reply:
Holy Cow, no wonder I couldn't find any documentation on that; it's pretty esoteric.  (I'm sure it's documented _somewhere_... but I couldn't find it.)

So now I have a new question:  it appears this is dependent on the ubuntu user login being named "judgeX" -- is that true?

And again, what happens if it is an Admin executing the submission?

Reply:
From above: "**If the account name is not "judge#", for example, it will use CPU 0, which is fine if only one judge is running (or admin) on the system. We usually only run one client application at-a-time on a machine, so that's ok**."